linked_list/single_linked_list.c

```diff
@@ -0,0 +1,186 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define ASSERT(x) printf("Assertion: " #x " = %d\n", (x))
+
+typedef int data_t;
+
+typedef struct node {
+  data_t data;
+  struct node *next;
+} node;
+
+#define CHECK_ALLOC(ptr)						\
+  if (!ptr) {								\
+    printf("Error allocating memory for %s in  %s: %d line", #ptr, __func__, __LINE__-1); \
+    exit(2);								\
+  }
+
+node* list_create(data_t data){
+  node* first = (node*)malloc(sizeof(node));
+  CHECK_ALLOC(first);
+  first->data = data;
+  first->next = NULL;
+  return first;
+}
+
+
+void list_free(node* first){
+  while (first != NULL)
+    {
+      node* tmp = first;
+      first = first->next;
+      free(tmp);
+    }
+}
+
+void list_append_element_end(node* list, int new_element){
+  node* new_node = (node*)malloc(sizeof(node)); 
+  CHECK_ALLOC(new_node); 
+  new_node->data = new_element;
+  new_node->next = NULL;
+
+  if (list == NULL){
+    list = new_node;
+  }
+  while (list->next != NULL)
+    list = list->next;
+
+  list->next = new_node;
+}
+
+
+int list_pop_end(node* list){
+  
+  node* prev_node = NULL;
+  while (list->next != NULL) {
+    prev_node = list;
+    list = list->next; 
+  } 
+  
+  if (prev_node)
+    prev_node->next = NULL;
+
+  int result = list->data; 
+  free(list);
+  return result;
+}
```

Comment:
I'd name it `list_pop`.

Some references:
* https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/pop
* https://docs.python.org/3/tutorial/datastructures.html#more-on-lists

Also... you don't have to fix that, but keep in mind this booms when `list == NULL`... (it doesn't handle such edge case) (I would say it is similar to `atoi`).
It could either set a global variable that would say "it errored" (such global would have to be cleared by the user of the function before calling it) (but globals are usually bad... also it wouldn't work with threads) or just return `-1` when `list == NULL` (and state it in some docstring/comment).